android/expoview/src/main/java/host/exp/exponent/utils/ToastHelper.kt

```diff
@@ -0,0 +1,44 @@
+package host.exp.exponent.utils
+
+import android.app.Application
+import android.os.Build
+import android.view.Gravity
+import android.widget.Toast
+import host.exp.exponent.ExponentManifest
+import host.exp.exponent.di.NativeModuleDepsProvider
+import org.json.JSONObject
+import javax.inject.Inject
+
+// TODO: Remove when we drop support for SDK 40
+
+object ToastHelper {
+  @Inject
+  var applicationContext: Application? = null
+
+  init {
+    NativeModuleDepsProvider.getInstance().inject(ToastHelper::class.java, this)
+  }
+
+  fun functionMayNotWorkOnAndroidRWarning(featureName: String, manifest: JSONObject?) {
+    try {
+      if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R) {
+        if (!ExponentManifest.isDebugModeEnabled(manifest)) {
+          return
+        }
+
+        applicationContext?.let {
+          val message = "$featureName may not work in Expo Go when you're using Android R.\nSee https://expo.fyi/android-r"
```

Comment:
```suggestion
          val message = "$featureName may not work in your app when you're using Android R.\nSee https://expo.fyi/android-r"
```
or does this really only apply to Expo Go and not standalone apps? 🤔